src/components/Auth.js

```diff
@@ -0,0 +1,207 @@
+  clientId: window._env_.KEYCLOAK_CLIENT_ID,
+  url: window._env_.KEYCLOAK_URL,
+  realm_access: { roles: [window._env_.KEYCLOAK_ROLE] },
+});
+
+const AuthContext = createContext(undefined);
+
+const useAuth = () => {
+  const state = useContext(AuthContext);
+
+  if (!state) {
+    throw new Error('context provider for AuthContext is missing');
+  }
+
+  return state;
+};
+
+const AuthProvider = (props) => {
+  const { children } = props;
+
+  const [logInOrOut, { isLoading: isAuthenticating, data, error }] = useMutation(
+    async (loggingIn) => {
+      if (loggingIn) {
+        const loggedIn = await keycloak.init({
+          checkLoginIframe: false,
+          enableLogging: true,
+          onLoad: 'login-required',
+        });
+
+        if (!loggedIn) {
+          await keycloak.login({ redirectUri: window.location.href });
+        }
+
+        const { kbToken: authorizationToken } = await api.post('/token', { keyCloakToken: keycloak.token }).request();
+        const { user } = jwt.decode(authorizationToken);
+
+        await keycloak.loadUserInfo();
+        // eslint-disable-next-line camelcase
+        const username = keycloak.userInfo?.preferred_username || user?.name;
+
+        return {
+          authenticationToken: keycloak.token,
+          authorizationToken,
+          isAuthenticated: true,
+          isAdmin: Boolean(user.groups.find(group => group.name === dbRoles.admin)),
+          hasWriteAccess: Boolean(user.groups.find(group => [dbRoles.admin, dbRoles.regular].includes(group.name))),
+          user,
+          username,
+        };
+      }
+
+      await keycloak.logout();
+      return undefined;
+    },
+  );
+
+  const { authorizationToken } = data ?? {};
+
+  useEffect(() => {
+    const unregister = fetchIntercept.register({
+      request: (fetchUrl, fetchConfig) => {
+        if (fetchUrl.startsWith(window._env_.API_BASE_URL)) {
+          const newConfig = { ...fetchConfig };
+
+          if (!newConfig.headers) {
+            newConfig.headers = {};
+          }
+          newConfig.headers.Authorization = authorizationToken;
+          return [fetchUrl, newConfig];
+        }
+        return [fetchUrl, fetchConfig];
+      },
+    });
+    return unregister;
+  }, [authorizationToken]);
+
+  const auth = useMemo(() => ({
+    login: () => {
+      if (!isAuthenticating) {
+        logInOrOut(true);
+      }
+    },
+    logout: () => {
+      if (!isAuthenticating) {
+        logInOrOut();
+      }
+    },
+    isAuthenticating,
+    error,
+    ...data || {},
+  }), [data, isAuthenticating, logInOrOut, error]);
+
+  return (
+    <AuthContext.Provider
+      value={auth}
+    >
+      {children}
+    </AuthContext.Provider>
+  );
+};
+AuthProvider.propTypes = {
+  children: PropTypes.node.isRequired,
+};
+
+const Centered = ({ children }) => (
+  <div>
+    <Box alignItems="center" display="flex" flexDirection="column" paddingTop="40px" width="100%">{children}</Box>
```

Comment:
Do these need to be in-line styles? we don't use in-line styles anywhere else

Reply:
they don't have to be written using mui, it just didn't seem worth add a folder and a stylesheet for a couple styles

Reply:
that makes sense. But i would like to keep it consistent with the other components so I think it is worth adding

Reply:
0c385fd